relayer/workers/beefyrelayer/fixture-data-logger.go

```diff
@@ -0,0 +1,136 @@
+		mmrProofItems = append(mmrProofItems, hex)
+	}
+
+	var signatures []string
+	for _, item := range msg.Signatures {
+		hex := hex.EncodeToString(item)
+		signatures = append(signatures, "0x"+hex)
+	}
+
+	var pubKeys []string
+	for _, item := range msg.ValidatorPublicKeys {
+		hex := item.Hex()
+		pubKeys = append(pubKeys, hex)
+	}
+
+	var pubKeyMerkleProofs [][]string
+	for _, pubkeyProof := range msg.ValidatorPublicKeyMerkleProofs {
+		var pubkeyProofS []string
+		for _, item := range pubkeyProof {
+			hex := "0x" + hex.EncodeToString(item[:])
+			pubkeyProofS = append(pubkeyProofS, hex)
+		}
+		pubKeyMerkleProofs = append(pubKeyMerkleProofs, pubkeyProofS)
+	}
+
+	wr.log.WithFields(logrus.Fields{
+		"msg.Commitment.BlockNumber":         msg.Commitment.BlockNumber,
+		"msg.Commitment.Payload":             payloadHex,
+		"msg.Commitment.ValidatorSetId":      msg.Commitment.ValidatorSetId,
+		"msg.Signatures":                     signatures,
+		"msg.ValidatorPositions":             msg.ValidatorPositions,
+		"msg.ValidatorPublicKeys":            pubKeys,
+		"msg.ValidatorPublicKeyMerkleProofs": pubKeyMerkleProofs,
+		"LatestMMRLeaf.ParentNumber":         msg.LatestMMRLeaf.ParentNumber,
+		"LatestMMRLeaf.ParentHash":           parentHashHex,
+		"LatestMMRLeaf.ParachainHeadsRoot":   parachainHeadsRootHex,
+		"LatestMMRLeaf.NextAuthoritySetId":   msg.LatestMMRLeaf.NextAuthoritySetId,
+		"LatestMMRLeaf.NextAuthoritySetLen":  msg.LatestMMRLeaf.NextAuthoritySetLen,
+		"LatestMMRLeaf.NextAuthoritySetRoot": nextAuthoritySetRootHex,
+		"hexEncodedLeaf":                     hexEncodedLeaf,
+		"hashedLeaf":                         hashedLeaf,
+		"mmrProofItems":                      mmrProofItems,
+	}).Info("Complete Signature Commitment transaction submitted")
+
+}
+
+// Keccak256 is the Keccak256 hashing method
+type Keccak256 struct{}
+
+// New creates a new Keccak256 hashing method
+func New() *Keccak256 {
+	return &Keccak256{}
+}
+
+// Hash generates a Keccak256 hash from a byte array
+func (h *Keccak256) Hash(data []byte) []byte {
+	hash := crypto.Keccak256(data)
+	return hash[:]
+}
+
+func (wr *BeefyEthereumWriter) GetFailingMessage(client ethclient.Client, hash common.Hash) (string, error) {
+	tx, _, err := client.TransactionByHash(context.Background(), hash)
+	if err != nil {
+		return "", err
+	}
+
+	from, err := types.Sender(types.NewEIP155Signer(tx.ChainId()), tx)
+	if err != nil {
+		return "", err
+	}
+
+	msg := ethereum.CallMsg{
+		From:     from,
+		To:       tx.To(),
+		Gas:      tx.Gas(),
+		GasPrice: tx.GasPrice(),
+		Value:    tx.Value(),
+		Data:     tx.Data(),
+	}
+
+	wr.log.WithFields(logrus.Fields{
+		"From":     from,
+		"To":       tx.To(),
+		"Gas":      tx.Gas(),
+		"GasPrice": tx.GasPrice(),
+		"Value":    tx.Value(),
+		"Data":     hex.EncodeToString(tx.Data()),
+	}).Info("Call info")
+
+	res, err := client.CallContract(context.Background(), msg, nil)
+	if err != nil {
+		return "", err
+	}
```

Comment:
What is the purpose of this function? It's in a logger file name, but does an actual call?

Reply:
yeh, it's just a call (basically simulates the tx without actually sending it), not an actual tx. added some info:
https://github.com/Snowfork/polkadot-ethereum/pull/442/commits/d33920b8b1feee369b1f0c26bef46e2c829e9c02